gitbook/graphics-transform.md

```diff
@@ -1,250 +1,228 @@
 
 ## Bounding boxes
 
-After transforming entities and drawing them, you might want to perform some computations using them e.g. checking for collisions. 
+After transforming entities and drawing them, you might want to perform some computations using them e.g. checking for collisions.
 
-SFML entities can give you their bounding box. The bounding box is the minimal rectangle that contains all points belonging to the entity, with sides aligned to the X and Y axes. 
+CrSFML entities can give you their bounding box. The bounding box is the minimal rectangle that contains all points belonging to the entity, with sides aligned to the X and Y axes.
 
 ![Bounding box of entities](images/graphics-transform-bounds.png)
 
-The bounding box is very useful when implementing collision detection: Checks against a point or another axis-aligned rectangle can be done very quickly, and its area is close enough to that of the real entity to provide a good approximation. 
+The bounding box is very useful when implementing collision detection: Checks against a point or another axis-aligned rectangle can be done very quickly, and its area is close enough to that of the real entity to provide a good approximation.
 
-```
-// get the bounding box of the entity
-sf::FloatRect boundingBox = entity.getGlobalBounds();
-
-// check collision with a point
-sf::Vector2f point = ...;
-if (boundingBox.contains(point))
-{
-    // collision!
-}
-
-// check collision with another box (like the bounding box of another entity)
-sf::FloatRect otherBox = ...;
-if (boundingBox.intersects(otherBox))
-{
-    // collision!
-}
-```
+```ruby
+# get the bounding box of the entity
+bounding_box = entity.global_bounds
 
-The function is named `getGlobalBounds` because it returns the bounding box of the entity in the global coordinate system, i.e. after all of its transformations (position, rotation, scale) have been applied. 
+# check collision with a point
+point = ...
 
-There's another function that returns the bounding box of the entity in its *local* coordinate system (before its transformations are applied): `getLocalBounds`. This function can be used to get the initial size of an entity, for example, or to perform more specific calculations. 
+if bounding_box.contains(point)
+    # collision!
+end
 
-## Object hierarchies (scene graph)
-
-With the custom transforms seen previously, it becomes easy to implement a hierarchy of objects in which children are transformed relative to their parent. All you have to do is pass the combined transform from parent to children when you draw them, all the way until you reach the final drawable entities (sprites, text, shapes, vertex arrays or your own drawables). 
+# check collision with another box (like the bounding box of another entity)
+other_box = ...
 
+if bounding_box.intersects(other_box)
```

Comment:
Well that was a huge mess in my code. No idea how this passed through unnoticed.